Reorganize the Spec (#140)
LibertyDSNP/spec: pages/Ethereum/Identity.md

> ---

Author: Updated this page to be specific for Ethereum. The chain agnostic content moved to pages/DSNP/Identity.md

name: Identity
route: /Ethereum/Identity
menu: "DSNP: Ethereum"
---

# Identity

## Purpose

1. Provide the interface for a DSNP identity.
1. Specify the delegation model and related interface.
1. Specify the ownership model and related interface.

> 1. Provide the list of EIPs that must be supported by a DSNP compatible identity.

Reviewer: Personally I always put the hyphen in compound adjectives like "DSNP-compatible", to make the binding clear and avoid any possible ambiguity.  Obviously, this comment would apply to all occurrences of that phrase in this spec, but I'm only noting it here, so as to contribute rather than annoy.

1. Provide the interface for a DSNP identity.
1. Specify the delegation model and related interface.
1. Specify the ownership model and related interface.
1. Provide the list of EIPs that must be supported by a DSNP compatible identity.

## Assumptions

* Identity is on Ethereum for Betanet.
* Identity will need to be moved from Betanet to Mainnet.

## Details

Identity is comprised of several parts:

* Ownership
* Delegation
* Identification
* Discovery

Additional constraints were placed upon any solution:

> * Pseudo Anonymity: A consistent identifier that defaults to disconnected from real world identity

Reviewer: "defaults to being disconnected" maybe?

Author: That's the old version. The new one is:
```markdown
* An identifier MUST default to being disconnected from a real world identity.
```

> The official [Identity Factory](/Ethereum/IdentityFactory) must be used for initial creation of a new identity.

Reviewer: Just curious: how would any observer of the chain be able to tell the difference between an identity that was created with the official Identity Factory and one that was compatibly created using some other means?  (Or maybe I'm misunderstanding what "Factory" means here?

Author: Ah here is that requirement. This is a mistake. It is not required.

```suggestion
The official [Identity Factory](/Ethereum/IdentityFactory) is provided for initial creation of a new identity.
```

## Delegation

> Delegation allows adding additional public keys that are allowed to sign announcements or perform other actions in addition to the owner public key(s).

Reviewer: "Delegation allows adding additional public keys (additional to the owner public key, that is) that are allowed to perform certain actions on behalf of the owner."  ?

I'm partly offering the rewrite to test my own understanding of what you wrote, but also (if my understanding was correct) to offer a clearer version.  Delegation is a precise thing; we need to make clear both exactly what it is and what its general role in DSNP is.  The latter concern is what motivated my "on behalf of" (as an alternative to "in addition to".  Although if "on behalf of" is not quite the right phrase then let's find something else.  One way or another, I think the spec needs to say something about what delegation *means* -- i.e., what someone who sees a delegated announcement can conclude about the upstream/original owning public key.

Author: That is correct. Here is an expanded version from your suggestion:

```suggestion
Delegation allows adding additional public keys, in addition to the owner public key(s).
These delegated keys are allowed to perform certain actions on behalf of the owner based on the delegate's "role".```
